src/pages/groupComparison/UpSet.tsx

```diff
@@ -0,0 +1,492 @@
+    groups: {
+        uid: string;
+        cases: string[];
+        name: string;
+    }[];
+    svgId?: string;
+    title?: string;
+};
+
+const PLOT_DATA_PADDING_PIXELS = 20;
+const DEFAULT_BOTTOM_PADDING = 10;
+const RIGHT_PADDING_FOR_LONG_LABELS = 50;
+const BAR_WIDTH = 20;
+
+@observer
+export default class UpSet extends React.Component<IUpSetrProps, {}> {
+
+    constructor(props: IUpSetrProps, context: any) {
+        super(props, context);
+    }
+
+    @observable.ref private container: HTMLDivElement;
+    @observable.ref tooltipModel: any | null = null;
+    @observable pointHovered: boolean = false;
+    private mouseEvents: any = this.makeMouseEvents();
+
+    @autobind
+    private containerRef(container: HTMLDivElement) {
+        this.container = container;
+    }
+
+    private makeMouseEvents() {
+        let disappearTimeout: Timer | null = null;
+        const disappearDelayMs = 250;
+
+        return [{
+            target: "data",
+            eventHandlers: {
+                onMouseOver: () => {
+                    return [
+                        {
+                            target: "data",
+                            mutation: (props: any) => {
+                                this.tooltipModel = props;
+                                this.pointHovered = true;
+
+                                if (disappearTimeout !== null) {
+                                    clearTimeout(disappearTimeout);
+                                    disappearTimeout = null;
+                                }
+
+                                return { active: true };
+                            }
+                        }
+                    ];
+                },
+                onMouseOut: () => {
+                    return [
+                        {
+                            target: "data",
+                            mutation: () => {
+                                if (disappearTimeout !== null) {
+                                    clearTimeout(disappearTimeout);
+                                }
+
+                                disappearTimeout = setTimeout(() => {
+                                    this.pointHovered = false;
+                                }, disappearDelayMs);
+
+                                return { active: false };
+                            }
+                        }
+                    ];
+                }
+            }
+        }];
+    }
+
+    @computed get uidToGroup() {
+        return _.reduce(this.props.groups, (acc, next) => {
+            acc[next.uid] = next.name;
+            return acc;
+        }, {} as { [id: string]: string });
+    }
+
+    @computed get groupLabels() {
+        return _.map(this.props.groups, group => group.name);
+    }
+
+    @computed get groupKeys() {
+        return _.keys(this.props.groups)
+    }
+
+    @computed get barSeparation() {
+        return 0.2 * this.barWidth;
+    }
+
+    @computed get barWidth() {
```

Comment:
i don't think any of the following getters need to be computed since they aren't doing any computation.

Reply:
true, I was using them from a existing component. I'll clean it